pkg/api/tests.go

```diff
@@ -132,3 +154,129 @@ func PrintTestsJSON(release string, w http.ResponseWriter, req *http.Request, cu
 		sort(req).
 		limit(req))
 }
+
+func PrintTestsJSONFromDB(release string, w http.ResponseWriter, req *http.Request, dbc *db.DB) {
+	tests := testsAPIResult{}
+	var filter *Filter
+
+	queryFilter := req.URL.Query().Get("filter")
+	if queryFilter != "" {
+		filter = &Filter{}
+		if err := json.Unmarshal([]byte(queryFilter), filter); err != nil {
+			RespondWithJSON(http.StatusBadRequest, w, map[string]interface{}{"code": http.StatusBadRequest, "message": "Could not marshal query:" + err.Error()})
+			return
+		}
+	}
+
+	// If requesting a two day report, we make the comparison between the last
+	// period (typically 7 days) and the last two days.
+	period := req.URL.Query().Get("period")
+	if period != "current" && period != "twoDay" {
```

Comment:
The UI sends `default` I think, not `current`.  It should probably allow a blank period, too, to mean default.

Reply:
Hmm weird, I see a couple refs to 'default' in the UI code, but I don't recall ever seeing that respected in the api server side code. I'll try to respect that broadly throughout with some util methods.

Reply:
I may have missed or removed period entirely in the jobs report, gonna fix that as well to respect it the same as above. I see the old jobs code did if twoDay, else default, so that is probably how it was working without ever calling out "default".